snuba/query/parser/expressions.py

```diff
@@ -61,44 +62,70 @@ def visit_function_name(self, node: Node, visited_children: Iterable[Any]) -> st
     def visit_column_name(self, node: Node, visited_children: Iterable[Any]) -> Column:
         return Column(None, None, node.text)
 
-    def visit_empty(
-        self, node: Node, visited_children: Iterable[Any]
-    ) -> Optional[Expression]:
-        return None
+    def visit_low_pri_op(self, node: Node, visited_children: Iterable[Any]) -> Any:
+        return node.text
```

Comment:
I think it would be best to resolve the operator here to something more abstract here than in the parent. So you make the visitor of the parent nodes independent of what the actual string is in the child.
Why not defining two enums, one for low pri operators and one for hi pri ones like 
```
class LowPriOperator(Enum):
   PLUS = "+"
   MINUS = "-"
```

Then in this visitor you resolve the string to the enum. Basically this code
```
if node.text == "+":
   return LowPriOperator.PLUS
.....
```

And instead of returning Any you return the enum. 
Now the parent node does not have to verify that the operator is valid, it just gets the enum which is already valid.

Reply:
Addressed.

Reply:
One note - in the classes, I had to use if - else structure logic, because if I used if - elif (where the operators are being checked in both statements), the type checker complained that there is no return statement.